### DISCUSSION TITLE
Best approach to have tags in supabase?

### DISCUSSION THREAD

**Discussion by malalecherocks** [supabase/supabase#8912] [General]
I have a table contact that could be 20M rows.

and I want to allow the user to add many tags to each contact: “man” “France” “toyota” … each contact can have up to 20 tags

i would like to be able to:

- query based on tags, with And and OR so I can filter to the user based on tags

What’s the best approach? A column with text array?(is possible in supabase?) a secondary table of tags? A column with full search text?

how would you do this? Any idea will be welcome

**Comment by GaryAustin1** [comment]
I would do a column with text array if these are somewhat free form tags.  Otherwise if limited might do an array of tag ids.   Then filtering is done using contains array operator which ands by default and can be used with or, OR you can use other array operators like overlap.
